src/nestor/models/notes_book.py

```diff
@@ -1,13 +1,51 @@
 from collections import UserDict
 
+class Note:
+    """Class representing a record for NotesBook."""
+
+    def __init__(self, title, content=None, tags=None):
+        """Initialize a new Note."""
+        self.title = title
+        self.tags = tags if tags else []  # Initialize tags
+        self.note_content = content if content else None  # Initialize note content
+
+    def add_tag(self, tag):
+        """Add a new tag to the note if it does not already exist."""
+        if tag not in self.tags:
+            self.tags.append(tag)
+
+    def delete_tag(self, tag):
+        """Delete a tag from the note if it exists."""
+        if tag in self.tags:
+            self.tags.remove(tag)
+
+    def change_content(self, content):
+        """Change the content of the note."""
+        self.note_content = content if content else None
+
+    def __str__(self):
+        """Return a string representation of the note."""
+        tags_str = ", ".join(self.tags) if self.tags else "No tags"
+        content_str = self.note_content if self.note_content else "No content"
+        return f"Title: {self.title}, Tags: [{tags_str}], Content: {content_str}"
+
 class NotesBook(UserDict):
-	"""
-	This class represents a book of notes.
-	"""
-	pass
+    """Class representing a NotesBook."""
 
-class Note:
-	"""
-	This class represents a note.
-	"""
-	pass
\ No newline at end of file
+    def add_note_record(self, note_record):
+        """Add a Note to the NotesBook."""
+        self.data[note_record.title] = note_record
+
+    def find(self, title):
+        """Find a Note by its title."""
+        return self.data.get(title, None)
+
+    def delete(self, title):
+        """Delete a Note from the NotesBook by its title."""
+        if title in self.data:
```

Comment:
`if title in self.data:` - is not needed I believe, because this method will silently work and handler will show `deleted successfully message` if such note was not found

Reply:
changed